### ISSUE TITLE
DOC502 false positive on `raise e`

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#20973]
### Summary

[`docstring-extraneous-exception` (DOC502)](https://docs.astral.sh/ruff/rules/docstring-extraneous-exception/) reports a false positive when an exception mentioned in the docstring is caught and explicitly reraised (i.e. `raise e` or `raise e from None`, assuming `e` is bound in the `except` clause and not rebound before it is raised). It correctly ignores implicit reraising (i.e. just `raise`) of a documented exception. The upstream implementation in pydoclint gets it right both ways. [Example](https://play.ruff.rs/ffc4e2ea-af0f-4ca4-a56e-ea2c40c373f8):
```console
$ cat >doc502.py <<'# EOF'
def f():
    """Do nothing.

    Raises
    ------
    OSError
        If the OS errors.
    """
    try:
        pass
    except OSError as e:
        raise e
# EOF

$ ruff --isolated check doc502.py --select DOC502 --preview --output-format concise -q
doc502.py:2:5: DOC502 Raised exception is not explicitly raised: `OSError`

$ pydoclint doc502.py
Skipping files that match this pattern: \.git|\.tox
doc502.py
🎉 No violations 🎉
```

### Version

ruff 0.14.1 (2bffef596 2025-10-16)
